src/pkg/services/m365/api/graph/errors.go

```diff
@@ -557,6 +563,16 @@ func parseODataErr(err error) oDataErr {
 	return ode
 }
 
+// hasResponseCode checks if the error is an ODataError and carries the provided
+// http response code.
+func (ode oDataErr) hasResponseCode(err error, httpResponseCode int) bool {
+	if !ode.isODataErr {
+		return false
+	}
+
+	return ode.Resp.StatusCode == httpResponseCode
```

Comment:
nit: could be simplified
```suggestion
	return ode.isODataErr && ode.Resp.StatusCode == httpResponseCode
```